scripts/build/termux_step_install_license.sh

```diff
@@ -60,29 +60,26 @@ termux_step_install_license() {
 						else
 							cp -f "${TERMUX_PKG_SRCDIR}/$FILE" "${TERMUX_PREFIX}/share/doc/${TERMUX_PKG_NAME}/copyright"
 						fi
-						# since this is a post-increment, (( 0 )) would be falsey
-						# thus `set -e` would kill the script on the first iteration
-						# using `true` prevents this
-						: $(( COUNTER++ ))
 					}
 				done
 				;;
 				# For the rest we can use a link to the generic license file
 				*) [[ -f "$TERMUX_SCRIPTDIR/packages/termux-licenses/LICENSES/${LICENSE}.txt" ]] && {
 						# the link target depends on the libc being used
-						if [[ "$TERMUX_PACKAGE_LIBRARY" == 'bionic' ]]; then
-							TO_LICENSE="../../LICENSES/${LICENSE}.txt"
-						elif [[ "$TERMUX_PACKAGE_LIBRARY" == 'glibc' ]]; then
-							TO_LICENSE="../../../../share/LICENSES/${LICENSE}.txt"
-						fi
+						case "$TERMUX_PACKAGE_LIBRARY" in
+							'bionic') TO_LICENSE="../../LICENSES/${LICENSE}.txt";;
+							'glibc')  TO_LICENSE="../../../../share/LICENSES/${LICENSE}.txt";;
+							*)        termux_error_exit "'$TERMUX_PACKAGE_LIBRARY' is not a supported libc";;
+						esac
 						if (( COUNTER )); then
-							ln -sf "$TO_LICENSE" "$TERMUX_PREFIX/share/doc/$TERMUX_PKG_NAME/copyright.$((COUNTER++))"
+							ln -sf "$TO_LICENSE" "$TERMUX_PREFIX/share/doc/$TERMUX_PKG_NAME/copyright.${COUNTER}"
 						else
 							ln -sf "$TO_LICENSE" "$TERMUX_PREFIX/share/doc/$TERMUX_PKG_NAME/copyright"
 						fi
 					}
 				;;
 			esac
+			(( ++COUNTER ))
```

Comment:
Like I said on matrix, counter increments should remain inside respective `if` after a new license is installed.

Reply:
Like https://github.com/TomJo2000/termux-packages-prs/blob/6aeb8b4fec9d209249b144385bd64c8c00b00a53/scripts/build/termux_step_install_license.sh#L63 and https://github.com/TomJo2000/termux-packages-prs/blob/6aeb8b4fec9d209249b144385bd64c8c00b00a53/scripts/build/termux_step_install_license.sh#L77

Reply:
We could go back to this, but `(( ++COUNTER ))` works just as well.

I think the crucial thing here is to decouple incrementing the counter from the specific case we get.
At the end of the `while` loop the counter ***always*** needs to be incremented so the next license is installed to the next index.
In hindsight I have no idea why I coupled the counter increment to the specific cases.

Reply:
Then you would get jumps in counter when `-f` file check fails and license isn't installed in current loop.

Reply:
Fair point.
We should probably exit out with a `Could not find any license file for $LICENSE` or something along those lines in that case.

Reply:
Maybe that's required for some package, I don't know, current condition is that at least one license needs to be installed. Maybe a warning could be printed instead.

Reply:
My standpoint is.
- If it's defined we defer to the license files explicitly defined by `TERMUX_PKG_LICENSE_FILE`

If `TERMUX_PKG_LICENSE` specifies multiple licenses, but no files:
- For the "author information" licenses,
https://github.com/termux/termux-packages/blob/cdec2239444e2844c755d39354823c42e6e7c866/scripts/build/termux_step_install_license.sh#L50-L54
Include all found license files from the search list.
https://github.com/termux/termux-packages/blob/cdec2239444e2844c755d39354823c42e6e7c866/scripts/build/termux_step_install_license.sh#L35-L43
- For the "generic licenses"
https://github.com/termux/termux-packages/blob/cdec2239444e2844c755d39354823c42e6e7c866/scripts/build/termux_step_install_license.sh#L70-L74
Link exactly one file for each of those licenses.
e.g. If a hypothetical package is licensed under:
  - `LGPL-3.0`, `GPL-3.0` and `ISC`
  we link in those licenses from `termux-licenses`

If we then have a package licensed under `GPL-1.0` for example,
which isn't in the `termux-licenses` package, that's an error
and we should error out because not all licenses are present and accounted for.
<sup>(And I'm rambling again... hopefully that at east made some sense)</sup>

Reply:
You didn't have to ramble, I already understood you. ;) Point is that I don't know if it's expected behavior for some package for its license file to not be found, maybe some package does not have a license file added in its source for one specific license, like in some termux app repos, license was only specified in README, so technically not an error. If no license was installed, then abortion would already happen.

Reply:
> Point is that I don't know if it's expected behavior for some package for its license file to not be found, maybe some package does not have a license file added in its source for one specific license, like in some termux app repos, license was only specified in README, so technically not an error.

My two cents on the matter is.
If it's listed as a `TERMUX_PKG_LICENSE` in the package build, then there has to be a license file for it.
Be that from the source repo or a generic one linked from `termux-licenses`.
For special cases there's always `TERMUX_PKG_LICENSE_FILE`.

`gtkwave` is a great example here.
https://github.com/termux/termux-packages/blob/e5d16a9f22cbc02c0fbe0ccc4a91887282f2562c/x11-packages/gtkwave/build.sh#L1-L4
GPL-2.0 is "generic" and gets linked in.
MIT is "specific" so needs to be copied from the source files.

This would be perfectly fine,
except we run into the current counter bug and the MIT license tries to copy through the dangling GPL-2.0 license symlink.
Thus as a workaround we need to specify the license files explicitly right now.

Reply:
> If it's listed as a TERMUX_PKG_LICENSE in the package build, then there has to be a license file for it.

Currently, license also won't get installed if its not in the `COMMON_LICENSE_FILES`.

> For special cases there's always TERMUX_PKG_LICENSE_FILE.

That could be done too, but would require that both variables be defined.

Reply:
> Currently, license also won't get installed if its not in the `COMMON_LICENSE_FILES`.

Which is expected and is where `TERMUX_PKG_LICENSE_FILE` comes in, for nonstandard license file names.

> That could be done too, but would require that both variables be defined.

Which is fine.

Reply:
>Which is expected and is where TERMUX_PKG_LICENSE_FILE comes in, for nonstandard license file names.

If `TERMUX_PKG_LICENSE_FILE` was set, then the `COMMON_LICENSE_FILES` `else` wouldn't have run anyways to miss license files.

Reply:
> If `TERMUX_PKG_LICENSE_FILE` was set, then the `COMMON_LICENSE_FILES` `else` wouldn't have run anyways to miss license files.

Yes correct.
If `TERMUX_PKG_LICENSE_FILE` is set we defer to the files specified there.
That's been the behavior since before the rewrite.

Reply:
So basically, checks should be added for both generic and non-generic licenses if they are missing if you wanna go that way, or neither. If this was an issue and abortion should always happen on a missing license, then why wasn't it done before by @Grimler91 and only at least one license file check was used instead. If there was no legitimate reason, then we could go with always abort way and fix packages that fail.

https://github.com/TomJo2000/termux-packages-prs/commit/b9db6b4de843f0eb05943f390c7ca9f37b1ba1a2

Reply:
To me erroring for missing license files seems like the most sensible way to handle them.
We want all specified licenses present and accounted for, and this to me seems like a sensible way to ensure that.
We'd also want to know what license is missing.

Reply:
I think we can close this discussion thread now that I finally got the point about needing to do the `COUNTER` increment for each found "specific" license file?

Reply:
Sure